File: BinarySearch.py
def binarySearch(arr, low, high, x):
    if high >= low:
        mid = (high + low) // 2
        if arr[mid] == x:
            return mid
        elif arr[mid] > x:
            return binarySearch(arr, low, mid - 1, x)
        else:
            return binarySearch(arr, mid + 1, high, x)
    else:
        return -1
#usage
my_list = list(map(int,input("Enter an array: ").split()))
target = int(input("Enter element to find: "))
result = binarySearch(my_list, 0, len(my_list)-1, target)

if result != -1:
    print(f"Element {target} found at index {result}")
else:
    print(f"Element {target} not found")

File: Kotlin/binarySearch.kt
fun binarySearch(arr: IntArray, target: Int): Int {
    var left = 0
    var right = arr.size - 1

    while (left <= right) {
        val mid = left + (right - left) / 2

        if (arr[mid] == target) {
            return mid // Target element found, return its index.
        } else if (arr[mid] < target) {
            left = mid + 1 
        } else {
            right = mid - 1 
        }
    }

    return -1 
}
